### DISCUSSION TITLE
Spacy SatterTextPlot Questions

### DISCUSSION THREAD

**Discussion by AnubhavCR7** [JasonKessler/scattertext#95] [Q&A]
Hello,
   I am currently doing a Sentiment Analysis and Topic modelling project, where I found the ScatterText plot very useful, for visualizing my results. So I am looking to build a dashboard where I can display the graph, after the code finishes running in the backend. But I had few questions in mind.

1. Can we modify the size of text displayed on the graph, based on assigned scaled f-score ?
Like terms with high scaled f-score should appear larger in the plot.

2. Are there any parameters or options that allows us to add Slide Ranger and Zoom in option in ScatterText plot ? (Like Plotly in Python)

3. I want to filter out terms appearing on the graph, based on two options --- one based on term frequency and the other based on top n terms, where n is entered by the user. For the first case, I know there is an option called 'minimum_term_frequency' based on which we can filter out terms whose frequency is less than a given threshold.  
But I am not sure how to do it for the second option. Like considering only the top  n  terms, arranged in descending order, based on their scaled f-score or frequency.

4. Can we save the plot in some format, other than HTML ? Maybe say in .png or .json format ? Because I am building the Dashboard using Plotly Dash library in Python. And I am finding it difficult to load HTML files using plotly.   PNG or JSON format might work. 

Any help is much appreciated. 
Regards.

**Comment by JasonKessler** [comment]
Except (3), the answer to all your questions is no. If you'd like, feel free to write these features into Scattertext and submit PRs.

For (3), you can figure out the terms you want to remove separately, and then use corpus.remove_terms(...) functions to remove them.

**Reply by AnubhavCR7** [reply]
So in the remove_terms() function, do we pass the list of terms that is to be excluded ? Because in the Source code that particular line is written like this :

corpus = corpus.remove_terms(set(corpus.get_terms()) - set(characteristic_scores.index))

Looks like characteristic_scores is a dataframe that is returned by the characteristic_scorer.get_scores(corpus) function. 

So how can we generate the terms programatically that we want to remove, or do we need to select them manually ?
Manually meaning scan through the document and select the terms that we want to remove.

**Reply by JasonKessler** [reply]
That's right. Pass a list or set of strings you'd like to remove. You can use corpus.get_term_freq_df() (as shown in the readme) to gather frequencies to figure out what terms you'd want to remove. 
Best of luck!

**Reply by AnubhavCR7** [reply]
Thanks ! Just another question, I see in the ScatterText plot, a column called "Characteristics" where different keywords in the Corpus are mentioned. Can you please explain what exactly "Characteristics" mean in this context ?
Regards.

**Reply by JasonKessler** [reply]
This is covered in the Readme. It's the rank difference between a corpus' term frequencies and a general list of English word frequencies. 
Locking  this thread since the original questions have been answered.
